### PR TITLE
feat: adds nav rail lab component

### PR DESCRIPTION
**Pull request by michaeltamaki-okta**
[OKTA-825118](https://oktainc.atlassian.net/browse/OKTA-825118)

## Summary

feat: adds nav rail lab component

Building out the presentation layer of the "App Switcher" https://odyssey.okta.design/latest/components/app-switcher/design-mla5Wn5G (now called "Nav Rail").
Figma: https://www.figma.com/design/zh7A9gjaDlI50Hctdvd1OB/Odyssey-Components?node-id=22141-142254&node-type=frame&t=Qc7MjiWUFNtOh9Vd-0

### Code structure

- [x] Organize code using a modular component architecture.
- [x] Use semantic HTML for clear and meaningful structure.
- [x] Break down complex components into smaller, reusable parts.

### Accessibility implementation

- [x] Apply ARIA attributes to indicate roles and relationships.
- [x] Test focus management and ensure interactive elements are keyboard accessible.
- [x] Provide visual alternatives for auditory elements.

### Internationalization (i18n) implementation

- [x] Separate any text content from the component's code for translation.
- [x] Test RTL support in Storybook

Screenshot 2024-11-15 at 10 19 30 AM

### Cross-browser compatibility

- [x] Ensure consistent behavior and appearance across browsers.

#### Chrome 
Screenshot 2024-11-15 at 10 14 34 AM
Screenshot 2024-11-15 at 10 14 45 AM

#### Safari
Screenshot 2024-11-15 at 10 15 13 AM
Screenshot 2024-11-15 at 10 15 21 AM

#### Firefox
Screenshot 2024-11-15 at 10 17 37 AM
Screenshot 2024-11-15 at 10 17 43 AM

### Documentation

- [x] Include code snippets, demos, and examples for various use cases.
- [x] Provide guidance on customization and handling special cases.

### Finalize code

- [ ] Interaction designers, UI eng and visual designers need to review that the Storybook code matches the intended Figma design
- [ ] Interaction designers need to confirm Storybook code matches Supernova
- [ ] Interaction and visual designers merge Figma and code at the same time
- [ ] Share updates to the [#design-system-team](https://okta.slack.com/archives/C02JL12CJPQ)

### QA

- [x] Include necessary unit tests to continuously verify component functionality
- [x] Screen reader test error/success states or message
- [x] Test edge cases and error scenarios to validate robustness

### Testing and validation

- [ ] Share work in [Craft](https://docs.google.com/document/d/1dqaEzKW3SlKccw2ILjX7y3LQY0toeNe9OuwpOs2XxcI/edit#heading=h.j8zakyclqze5) to get feedback from the Odyssey team
- [ ] Share work to get feedback from pillar design captains or product designers in [#design-team](https://okta.slack.com/archives/G7REPJFUY) as customer feedback is essential to help us validate our assumptions, identify pain points, and improve our user experience

### CODE REVIEW

File: packages/odyssey-react-mui/src/labs/NavRail/OktaAura.tsx [hunk 1]
```diff
@@ -10,14 +10,20 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
+import styled from "@emotion/styled";
 import { memo } from "react";
 
+const OktaAuraSvgComponent = styled("svg")(() => ({
```
Comment (bryancunningham-okta) [inline packages/odyssey-react-mui/src/labs/NavRail/OktaAura.tsx:16]: @michaeltamaki-okta I think we can use the svg that currently lives in the `SideNav` folder. We can just wrap it in a styled component and apply these sizes to it. Rather than duplicate the SVG with a different size.

File: packages/odyssey-react-mui/src/labs/NavRail/NavRailApp.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    borderRadius: odysseyDesignTokens.BorderRadiusTight,
+    transition: `background-color ${odysseyDesignTokens.TransitionDurationMain}`,
+
+    "&:hover, &:focus": {
+      backgroundColor: odysseyDesignTokens.HueNeutral50,
+    },
+  }),
+);
+
+const NavRailAppImgComponent = styled("img")(() => ({
```
Comment (bryancunningham-okta) [inline packages/odyssey-react-mui/src/labs/NavRail/NavRailApp.tsx:61]: IMHO, It would be great if we could render the SVGs inline, rather than using images here

**Comment by michaeltamaki-okta** [comment]
Closing in favor of https://github.com/okta/odyssey/pull/2416
